superdesk/core/resources/service.py

```diff
@@ -564,4 +567,58 @@ def filter_ignore_fields(d, fields):
         return h.hexdigest()
 
 
+class AsyncCacheableService(AsyncResourceService[ResourceModelType]):
+    """
+    Handles caching for the resource, will invalidate on any changes to the resource.
+
+    Attributes:
+        resource_name (str): The name of the resource this service handles.
+        cache_lookup (dict): A dictionary to specify custom lookup parameters for caching.
+    """
+
+    cache_lookup = {}
+
+    @property
+    def cache_key(self) -> str:
+        return "cached:{}".format(self.resource_name)
+
+    def get_cache(self) -> Any:
+        """
+        Retrieve the cached value from Flask's `g` object if available.
+        """
+        return getattr(g, self.cache_key, None)
+
+    def set_cache(self, value: Any) -> None:
+        """
+        Set the cached value in Flask's `g` object.
+        """
+        setattr(g, self.cache_key, value)
+
+    async def get_cached(self) -> List[Dict[str, Any]]:
```

Comment:
It would be good to document the methods of this class, and at least add API reference to the [docs](https://github.com/superdesk/superdesk-core/blob/async/docs/core/resource_management.rst?plain=1#L59)

Reply:
done https://superdesk--2688.org.readthedocs.build/en/2688/